src/localmodeling.jl

```diff
@@ -6,47 +6,50 @@ export LocalAverageModel,LocalLinearModel,LocalPolynomialModel
 export TSP
 export MSE1,MSEp
 export estimate_param
-abstract type AbstractLocalModel end
 
+"""
+    AbstractLocalModel
+Supertype of methods for making a prediction `y` for the query `q` given it's nearest
+neighbors `xnn` and their respective images `ynn`. The distances between `q` and the `xnn`
+are given in `dists` to allow for a weighted model.
+
+Concrete subtypes:
+  * `LocalAverageModel(n::Int)`   : Compute a weighted average over the `ynn`.
+  * `LocalLinearModel(n::Int, f::F)` : Compute a weighted linear regression over
```

Comment:
I don't think we should allow `LocalLinearModel(n::Int, f::F)` with `f` a function. We should instead have:

* `LocalLinearModel(n::Int, μ::Real)`
* `LocalLinearModel(n::Int, s1::Real, s2::Real)`

We can't handle "any" function, and thus you cannot expose the choice of "any" function to a general API.

Since one case has 1 parameter and the other one 2, we can distinguish.